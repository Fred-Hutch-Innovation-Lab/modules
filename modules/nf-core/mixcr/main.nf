process MIXCR {
    tag "$meta.id"
    label 'process_medium'

    // TODO nf-core: See section in main README for further information regarding finding and adding container addresses to the section below.
    conda "${moduleDir}/environment.yml"
    // container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
    //     'https://depot.galaxyproject.org/singularity/mixcr:4.7.0--hdfd78af_0':
    //     'biocontainers/mixcr:4.7.0--hdfd78af_0' }"

    container 'ghcr.io/milaboratory/mixcr/mixcr:4.7.0-164-develop'

    input:
    tuple val(meta), path(reads)
    val preset
    val species
    env MI_LICENSE

    output:
    tuple val(meta), path("*clones*.tsv"), emit: clones
    tuple val(meta), path("*.txt"),        emit: reports
    tuple val(meta), path("*.clns"),       emit: clns, optional: true
    tuple val(meta), path("*.vdjca"),      emit: vdjca, optional: true
    path "versions.yml",                   emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def java_mem = task.memory ? "-Xmx${task.memory.toGiga()}g" : ''
    """
    mixcr $java_mem analyze \\
        $preset \\
        --species $species \\
        $args \\
        --threads $task.cpus \\
        ${reads} \\
        $prefix

    cat <<-END_VERSIONS > versions.yml
    "$task.process":
        mixcr: \$(mixcr -v 2>&1 | sed -n '1p' | sed -E 's/MiXCR v([0-9\\.]+).*/\1/' || true)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    mixcr -v 
    touch ${prefix}.clones_TRA.tsv
    touch ${prefix}.clones_TRB.tsv
    touch ${prefix}.clns
    touch ${prefix}.report.txt

    cat <<-END_VERSIONS > versions.yml
    "$task.process":
        mixcr: \$(mixcr -v 2>&1 | sed -n '1p' | sed -E 's/MiXCR v([0-9\\.]+).*/\1/' || true)
    END_VERSIONS
    """
}
