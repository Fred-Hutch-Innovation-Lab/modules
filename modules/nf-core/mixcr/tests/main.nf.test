// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test mixcr
nextflow_process {

    name "Test Process MIXCR"
    script "../main.nf"
    process "MIXCR"

    tag "modules"
    tag "modules_nfcore"
    tag "mixcr"

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used
    test("genomics - homo_sapiens - 10xgenomics - cellranger - fastq") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/10xgenomics/cellranger/5k_cmvpos_tcells/fastqs/vdj/subsampled_5k_human_antiCMV_T_TBNK_connect_VDJ_S1_L001_R1_001.fastq.gz', checkIfExists: true), file(params.modules_testdata_base_path + '/genomics/homo_sapiens/10xgenomics/cellranger/5k_cmvpos_tcells/fastqs/vdj/subsampled_5k_human_antiCMV_T_TBNK_connect_VDJ_S1_L001_R2_001.fastq.gz', checkIfExists: true) ]
                ]
                input[1] = "10x-vdj-tcr"
                input[2] = "hsa"
                input[3] = "E-BSBNBYVCBBIAGVCMOQMDQUATLQZUDKXBIPAARSDTDWUVMVRX"
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                // { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
                // See https://nf-co.re/docs/contributing/tutorials/nf-test_assertions for more information and examples.
            )
        }

    }

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used but keep the " - stub" suffix.
    test("10xgenomics vdj - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/10xgenomics/cellranger/5k_cmvpos_tcells/fastqs/vdj/subsampled_5k_human_antiCMV_T_TBNK_connect_VDJ_S1_L001_R1_001.fastq.gz', checkIfExists: true), file(params.modules_testdata_base_path + '/genomics/homo_sapiens/10xgenomics/cellranger/5k_cmvpos_tcells/fastqs/vdj/subsampled_5k_human_antiCMV_T_TBNK_connect_VDJ_S1_L001_R2_001.fastq.gz', checkIfExists: true) ]
                ]
                input[1] = "10x-vdj-tcr"
                input[2] = "hsa"
                input[3] = "E-BSBNBYVCBBIAGVCMOQMDQUATLQZUDKXBIPAARSDTDWUVMVRX"
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                // { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
            )
        }

    }

}
