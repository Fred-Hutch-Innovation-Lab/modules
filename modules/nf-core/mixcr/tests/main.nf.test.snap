{
    "10xgenomics vdj - fastqs - stub": {
        "content": [
            [
                "versions.yml:md5,41e6a626019639ed2c65f6926898c5fc"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-06-12T16:25:23.202221"
    },
    "10xgenomics vdj - fastq": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.clones.tsv:md5,bdf3ad0dfd9b3fee124bfa80e268e534"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    [
                        "test.alignments.vdjca:md5,9d9aea1fff4b5a95098ba369cc1e7251",
                        "test.extended.vdjca:md5,e95e9e676bdf9406a8273b964aeea37d",
                        "test.refined.vdjca:md5,a2e0929f02b4b9a9fd4cf49caa60cae4"
                    ]
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    [
                        "test.assembledCells.clns:md5,90698bdae65572bc22f3bcc3f0906686",
                        "test.contigs.clns:md5,dab59a29c7d5ad5b4a2d74bbcd00e3a6"
                    ]
                ]
            ],
            [
                "versions.yml:md5,41e6a626019639ed2c65f6926898c5fc"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-06-12T16:25:13.956998"
    }
}